x/ibc/04-channel/keeper/keeper.go

```diff
@@ -160,3 +160,28 @@ func (k Keeper) GetAllChannels(ctx sdk.Context) (channels []types.Channel) {
 	})
 	return channels
 }
+
+// LookupModuleByChannel will return the IBCModule along with the capability associated with a given channel defined by its portID and channelID
+func (k Keeper) LookupModuleByChannel(ctx sdk.Context, portID, channelID string) (string, capability.Capability, bool) {
+	capOwners, ok := k.scopedKeeper.GetOwners(ctx, ibctypes.ChannelCapabilityPath(portID, channelID))
+	if !ok {
+		return "", nil, false
+	}
+	if len(capOwners.Owners) < 2 {
+		panic("more than one module has bound to this port; currently not supported")
+	}
+	// the module that owns the channel, for now, is the only module owner that isn't "ibc"
+	var module string
+	if capOwners.Owners[0].Module == "ibc" {
+		module = capOwners.Owners[1].Module
+	} else {
+		module = capOwners.Owners[0].Module
+	}
+
+	cap, found := k.scopedKeeper.GetCapability(ctx, ibctypes.ChannelCapabilityPath(portID, channelID))
+	if !found {
+		return "", nil, false
+	}
+	return module, cap, true
```

Comment:
nit, you can avoid duplication by abstracting this part and passing the path:

```go
func (k ScopedKeeper) LookupModule(ctx sdk.Context, path string) (string, capability.Capability, bool) {
	capOwners, ok := k.GetOwners(ctx, path)
	if !ok {
		return "", nil, false
	}
	// For now, we enforce that only IBC and the module bound to port can own the capability
	// while future implementations may allow multiple modules to bind to a port, currently we
	// only allow one module to be bound to a port at any given time
	if len(capOwners.Owners) != 2 {
		panic("more than one module has bound to this port; currently not supported")
	}
	// the module that owns the port, for now, is the only module owner that isn't "ibc"
	var module string
	if capOwners.Owners[0].Module == "ibc" {
		module = capOwners.Owners[1].Module
	} else {
		module = capOwners.Owners[0].Module
	}
	cap, found := k.GetCapability(ctx, path)
	if !found {
		return "", nil, false
	}
	return module, cap, true
}
```
 
and then:

```go
// LookupModuleByChannel will return the IBCModule along with the capability associated with a given channel defined by its portID and channelID
func (k Keeper) LookupModuleByChannel(ctx sdk.Context, portID, channelID string) (string, capability.Capability, bool) {
	return k.scopedKeeper.LookupModule(ctx, ibctypes.ChannelCapabilityPath(portID, channelID))
}
```